drivers/sensor/icm42605/icm42605.c

```diff
@@ -0,0 +1,392 @@
+	}
+
+	return 0;
+}
+
+static int icm42605_attr_get(struct device *dev,
+			     enum sensor_channel chan,
+			     enum sensor_attribute attr,
+			     struct sensor_value *val)
+{
+	struct icm42605_data *drv_data = dev->data;
+
+	if (val == NULL) {
+		LOG_ERR("Attr set : val is null");
+		return -EINVAL;
+	}
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+	case SENSOR_CHAN_ACCEL_Y:
+	case SENSOR_CHAN_ACCEL_Z:
+	case SENSOR_CHAN_ACCEL_XYZ:
+		if (attr == SENSOR_ATTR_SAMPLING_FREQUENCY) {
+			val->val1 = drv_data->accel_hz;
+		} else if (attr == SENSOR_ATTR_FULL_SCALE) {
+			val->val1 = drv_data->accel_sf;
+		} else {
+			LOG_ERR("Not supported ATTR");
+			return -EINVAL;
+		}
+
+		break;
+	case SENSOR_CHAN_GYRO_X:
+	case SENSOR_CHAN_GYRO_Y:
+	case SENSOR_CHAN_GYRO_Z:
+	case SENSOR_CHAN_GYRO_XYZ:
+		if (attr == SENSOR_ATTR_SAMPLING_FREQUENCY) {
+			val->val1 = drv_data->gyro_hz;
+		} else if (attr == SENSOR_ATTR_FULL_SCALE) {
+			val->val1 = drv_data->gyro_sf;
+		} else {
+			LOG_ERR("Not supported ATTR");
+			return -EINVAL;
+		}
+
+		break;
+
+	default:
+		LOG_ERR("Not support");
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int icm42605_data_init(struct icm42605_data *data)
+{
+	data->accel_x = 0;
+	data->accel_y = 0;
+	data->accel_z = 0;
+	data->temp = 0;
+	data->gyro_x = 0;
+	data->gyro_y = 0;
+	data->gyro_z = 0;
+	data->accel_hz = 10;
+	data->gyro_hz = 10;
+
+	data->accel_sf = ACCEL_FS_16G;
+	data->gyro_sf = GYRO_FS_2000DPS;
+
+	data->tap_en = false;
+	data->sensor_started = false;
+
+	return 0;
+}
+
+
+static const struct sensor_driver_api icm42605_driver_api = {
+	.trigger_set = icm42605_trigger_set,
+	.sample_fetch = icm42605_sample_fetch,
+	.channel_get = icm42605_channel_get,
+	.attr_set = icm42605_attr_set,
+	.attr_get = icm42605_attr_get,
+};
+
+int icm42605_init(struct device *dev)
+{
+	struct icm42605_data *drv_data = dev->data;
+
+	drv_data->spi = device_get_binding(DT_INST_BUS_LABEL(0));
+
+	spi_cs.gpio_dev = device_get_binding(DT_INST_SPI_DEV_CS_GPIOS_LABEL(0));
+	spi_cs.gpio_pin = DT_INST_SPI_DEV_CS_GPIOS_PIN(0);
+	spi_cs.gpio_dt_flags = DT_INST_SPI_DEV_CS_GPIOS_FLAGS(0);
+	spi_cs.delay = 0U;
+	spi_cfg.cs = &spi_cs;
+
+	icm42605_spi_init(drv_data->spi, &spi_cfg);
+	icm42605_data_init(drv_data);
+
+	k_msleep(100);
+
+	icm42605_sensor_init(dev);
+
+	drv_data->accel_sensitivity_shift = 14 - 3;
+	drv_data->gyro_sensitivity_x10 = icm42605_gyro_sensitivity_x10[3];
+
+	if (icm42605_init_interrupt(dev) < 0) {
+		LOG_ERR("Failed to initialize interrupts.");
+		return -EIO;
+	}
+
+	LOG_DBG("Initialize interrupt done");
+
+	return 0;
+}
+
+
+static struct icm42605_data icm42605_driver;
+static const struct icm42605_config icm42605_cfg = {
+	.spi_label = DT_INST_SPI_DEV_CS_GPIOS_LABEL(0),
+	.spi_addr = DT_INST_REG_ADDR(0),
+	.int_label = DT_INST_GPIO_LABEL(0, int_gpios),
+	.int_pin =  DT_INST_GPIO_PIN(0, int_gpios),
+	.int_flags = DT_INST_GPIO_FLAGS(0, int_gpios),
+};
+
+DEVICE_AND_API_INIT(icm42605, DT_INST_LABEL(0),
```

Comment:
Now that it's possible I think it's better to create multi-instance drivers. You can search for `DT_INST_FOREACH_STATUS_OKAY` e.g. https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/kscan/kscan_ft5336.c#L275

Reply:
I will implement this feature at next time

Reply:
OK I already implemented and tested, I will commit soon.

Reply:
@gmarull Can you please review latest implementation?
I followed multi instance driver like link what you shared.